nixos/modules/services/editors/vim.nix

```diff
@@ -0,0 +1,33 @@
+{ config, lib, pkgs, ...}:
+
+with lib;
+
+let
+
+  cfg = config.services.vim;
+
+in {
+
+  options.services.vim =  {
+    defaultEditor = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = "Set VIM to be the default editor";
+    };
+  };
+
+    config = mkIf cfg.defaultEditor {
+      environment.systemPackages = [ pkgs.vim ];
```

Comment:
There are a bunch of different versions of vim in nixpkgs, can you include an option for the package that should be used.